alphadia/workflow/config.py

```diff
@@ -563,30 +82,219 @@ def update(self, experiments: list["Config"], print_modifications: bool = True):
+                experiment_config.to_dict(),
+                tracking_dict,
+                experiment_config.experiment_name,
+            )
+
+        self.config = current_config
+
+        if do_print:
+            try:
+                self._pretty_print(current_config, default_config, tracking_dict)
+            except Exception as e:
+                logger.warning(f"Could not print config: {e}")
+                logger.info(f"{(yaml.dump(current_config))}")
+
+    @staticmethod
+    def _pretty_print(config: dict, default_config: dict, tracking_dict: dict) -> None:
         """
+        Pretty print a configuration dictionary in a tree-like structure.
 
-        # The translated config contains the source of the modifications on leaf nodes
-        translated_config = translate_config(
-            copy.deepcopy(self.config), self.experiment_name
+        Args:
+            config: The configuration dictionary to print
+            default_config: The default configuration dictionary to print
+            tracking_dict: A dictionary with the same structure as config, whose leaf values contain the experiment name that last updated the value
+        """
+
+        _pretty_print(
+            config, default_config=default_config, tracking_dict=tracking_dict
         )
 
-        if len(experiments) > 0:
-            translated_experiments = []
-            for config in experiments:
-                translated_experiments.append(
-                    translate_config(
-                        copy.deepcopy(config.config), config.experiment_name
-                    )
+
+def _update(
+    target_config: dict,
+    update_config: dict,
+    tracking_dict: dict,
+    experiment_name: str,
+) -> None:
+    """
+    Recursively update target_dict in-place with values from update_dict, following specific rules for different types.
+
+    For each value that gets updated, the corresponding value in tracking_dict is updated with experiment_name.
+
+    Args:
+        target_config: The config dictionary to be modified
+        update_config: The config dictionary containing update values
+        tracking_dict: A dictionary of nested dictionaries.
+            If a value target_config gets overwritten, the same value in tracking_dict will be overwritten with `experiment_name`.
+        experiment_name: The name of the current experiment
+
+    Notes:
+        - Nested dictionaries are recursively updated
+        - Only updates existing keys (adding new keys not allowed)
+        - lists are always overwritten
+
+    Raises:
+        ValueError in these cases:
+        - a key is not found in the target_config
+        - the type of the update value does not match the type of the target value
+        - an item is not found in the target_config
+    """
+    for key, update_value in update_config.items():
+        if key not in target_config:
+            raise ValueError(f"Key not found in target_config: '{key}'")
```

Comment:
Should we make this more explicit? 
Something like: Updating configs does not allow defining new keys.